synapse/storage/data_stores/main/roommember.py

```diff
@@ -868,6 +868,37 @@ def get_membership_from_event_ids(
             desc="get_membership_from_event_ids",
         )
 
+    async def is_local_host_in_room_ignoring_users(
+        self, room_id: str, ignore_users: Collection[str]
+    ) -> bool:
+        """Checks if the local server is in the room, ignoring membership of
+        the given users.
+        """
+
+        clause, args = make_in_list_sql_clause(
+            self.database_engine, "user_id", ignore_users
+        )
+
+        sql = """
+            SELECT 1 FROM local_current_membership
+            WHERE
+                room_id = ? AND membership = ?
+                AND NOT (%s)
+                LIMIT 1
+        """ % (
+            clause,
+        )
```

Comment:
local_current_membership is populated in a bg update, so this needs to be cunninger.

Reply:
It isn't actually, afaict.

Reply:
https://github.com/matrix-org/synapse/blob/master/synapse/storage/data_stores/main/schema/delta/57/local_current_membership.py

Reply:
ah fair enough